client/state/ui/editor/reducer.js

```diff
@@ -21,8 +21,10 @@ import contactForm from './contact-form/reducer';
  */
 export function postId( state = null, action ) {
 	switch ( action.type ) {
-		case EDITOR_POST_ID_SET:
+		case EDITOR_START:
 			return action.postId;
+		case POST_SAVE_SUCCESS:
+			return state === action.postId ? action.savedPost.ID : state;
```

Comment:
I guess we need this condition since other posts could potentially be saved in the background that aren't related to the one we're editing? I guess it's okay, since we're effectively ensuring that the ID is kept in sync after a save (especially for draft -> saved post).